java/src/org/openqa/selenium/grid/node/config/SessionCapabilitiesMutator.java

```diff
@@ -215,6 +217,14 @@ private Map<String, Object> mergeFirefoxOptions(
         Map<String, Object> logLevelMap = (Map<String, Object>) value;
         toReturn.put("log", logLevelMap);
       }
+
+      if (name.equals("binary") && !stereotypeOptions.containsKey("binary")) {
+        toReturn.put(name, value);
+      }
```

Comment:
We intentionally decided to not do this one because the assumption is that the Grid is already configured and a test should not determine the browser driver binary location.

Reply:
Hi Diego!

There reason I've added this part is mainly because there is the same one for Chromium in the existing implementation -- [link](https://github.com/SeleniumHQ/selenium/blob/trunk/java/src/org/openqa/selenium/grid/node/config/SessionCapabilitiesMutator.java#L158)

Also, I think this condition covers the case when binary wasn't configured from grid to allow to set it from tests.

Anyway, I'm OK with removing it from my PR as it doesn't directly affect our case - let me know.

Reply:
I missed that. OK. Good point. Thank you.